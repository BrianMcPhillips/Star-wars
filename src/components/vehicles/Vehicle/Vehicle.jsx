import React from 'react';
import PropTypes from 'prop-types';

const Vehicle = ({ name, model, url }) => (
  <>
    <h1>{name}</h1>
    <h4>{model}</h4>
  </>
);

Vehicle.propTypes = {
  name: PropTypes.string.isRequired,
  model: PropTypes.string.isRequired
};

export default Vehicle;
