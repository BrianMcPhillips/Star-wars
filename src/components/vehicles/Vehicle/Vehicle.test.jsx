import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Vehicle from './Vehicle';


describe('Vehicle component', () => {
  afterEach(() => cleanup());
  it('renders Vehicle', () => {
    const { asFragment } = render(
      <Vehicle 
        name="Sand Crawler"
        model="Digger Crawler"
        url="https://swapi.dev/api/vehicles/4/"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
