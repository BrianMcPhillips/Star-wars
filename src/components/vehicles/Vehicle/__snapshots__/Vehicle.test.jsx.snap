// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicle component renders Vehicle 1`] = `
<DocumentFragment>
  <h1>
    Sand Crawler
  </h1>
  <h4>
    Digger Crawler
  </h4>
</DocumentFragment>
`;
